module/events/finalization_actor.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/tracker"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+// OnBlockFinalized is invoked when a new block is finalized. It is possible that
+// blocks will be skipped.
+type OnBlockFinalized func(block *model.Block) error
+
+// FinalizationActor is an event responder worker which can be embedded in a component
+// to simplify the plumbing required to respond to block finalization events.
+// This worker is designed to respond to a newly finalized blocks on a best-effort basis,
+// meaning that it may skip blocks when finalization occurs more quickly.
+// CAUTION: This is suitable for use only when the handler can tolerate skipped blocks.
+type FinalizationActor struct {
```

Comment:
I have a soft preference for the `FinalizationActor` to implement the `FinalizationConsumer` interface

Reply:
Done in https://github.com/onflow/flow-go/pull/4187/commits/bff4663ef672e2e7eea3ef50f8e69dea09fe9e8f.